container-data/etc/nginx/conf.d/default.conf

```diff
@@ -2,7 +2,15 @@ server {
     listen 80;
 
     index index.php index.html index.htm;
-    root /usr/share/nginx/budget/public;
+    root /app/public;
+
+    set $is_https "";
+    if ($http_x_forwarded_proto ~ https) {
+        set $is_https true;
+    }
```

Comment:
What is this supposed to do? Detect whether or not to use HTTPS as opposed to HTTP?

Reply:
I'm actually making an assumption here and I'm not really sure if I should even do that. The convention tends to be that in Docker/container-land all containers are behind some kind of load balancer/reverse proxy that handles stuff like the actual routing and also TLS termination. The container itself just does plain HTTP. 

What this does, it will set an empty ``$is_https`` variable, if it detects a ``X-Forwarded-Proto`` header from the proxy containg the value ``https`` it will set ``$is_https`` to ``true``.

Later in the PHP-FPM section it will check whether or not ``$is_https`` has been set or not and based on that it will set the HTTPS environmental variable. 

I'm not really into this PHP and more specific Laravel thing but it seems to be that Laravel checks for this header and alter its logic based on that, i.e. the URL generator will create URLs with https as scheme, et cetera. Am I correct here?